core/blockchain.go

```diff
@@ -407,7 +407,7 @@ func (bc *BlockChain) FastSyncCommitHead(hash common.Hash) error {
 
 // ShardID returns the shard Id of the blockchain.
 func (bc *BlockChain) ShardID() uint32 {
-	return uint32(bc.chainConfig.ChainID.Int64())
+	return bc.CurrentBlock().ShardID()
```

Comment:
This is hacky. Please add a TODO to have a better solution.

Reply:
I think this is the good way, since the shardID is dynamically changing after resharding. But let me what do youthink

Reply:
This is representing the blockchain. Blockchain itself can only have one ShardID

Reply:
Added todo list